src/commands/shared.ts

```diff
@@ -187,10 +187,19 @@ export const provisionRequest = async (
     throw "Public key mismatch. Please revoke the request and try again.";
   }
 
-  return { request: provisionedRequest, publicKey, privateKey };
+  const posixAccount =
+    provisionedRequest.permission.spec.type === "gcloud"
+      ? await importSshKey(publicKey, { debug: args.debug })
+      : undefined;
+  const linuxUserName = posixAccount?.username;
```

Comment:
It feels a little unexpected to me here to return the linux username from without any mention that it's GCP specific in the output. Since this is happening last could we move it to `requestToSsh`? Is there a disadvantage to running this code in `requestToSsh` on line 214?